### ISSUE TITLE
IgnoreList

### ISSUE THREAD

**Issue by jaredreich** [Librevault/librevault#33]
Amazing work on librevault so far.

This feature would have the same functionality as BitTorrent Sync's IgnoreList where it includes default filetypes to ignore, and allows you to add your old filetypes and directories to ignore.

**Comment by jaredreich** [comment]
I've created ```~/Library/Preferences/Librevault/folders.conf```
With this JSON:
```json
{
	"ignore_paths": ["test"]
}
```
And it doesn't work when creating a new folder called ```test```. It still syncs.

**Comment by GamePad64** [comment]
I've just discovered some mistakes in the blog. Actually, the blog page is not the right place for deep-technical articles. I've created a [GitHub wiki](https://github.com/Librevault/librevault/wiki/Librevault-Configuration) page for configuration.

First, the correct location for OSX is `~/Library/Application Support/Librevault`.
Second, the folder.conf format is a bit different: it is an array of objects. And every object must contain "secret" and "path". Also, this object may contain additional fields, like "ignore_paths".

And, regarding your "test" path: it will not work, because the daemon lacks the support of wildcards in paths and only supports ECMAScript notation, applied to full path. Your example with "test" folder should be written as `^test(?:\/(?:.*))?`. This is not convenient at all.

It is a serious issue, and I've created #36 to track it.

**Comment by jaredreich** [comment]
Thanks for the detailed response!

On a complete side note, you I sent you an email today to your gmail to get you to take a look at reddit since Librevault is a trending post right now. Perhaps you should answer some of the questions folks have there.

https://www.reddit.com/r/programming/comments/4x1qbu/librevault_open_source_zeroknowledge_peer2peer/

**Comment by GamePad64** [comment]
Thanks, I've seen your email and typing the replies right now :+1:

**Comment by jaredreich** [comment]
Cool. Congrats again on your great work on Librevault. Hope to see it grow and thrive.

**Comment by jaredreich** [comment]
Back to the ignoring of files/folders. If possible you should make it exactly like BitTorrent Sync, where there is a .IgnoreList file within the synced folder where you can list all the files/folders to ignore.

Cheers!

**Comment by GamePad64** [comment]
Yes, I am thinking on that, too. It is the same approach as `.ignore` files in Git, Mercurial, Syncthing and BTSync. I think, it needs to be implemented in the nearest releases. Created #37

**Comment by jaredreich** [comment]
Exactly!

**Comment by jaredreich** [comment]
watch out for questions/comments here too:
https://news.ycombinator.com/item?id=12261409
